### PR TITLE
Conn 1979

### PR DESCRIPTION
**Pull request by ptambellini01**

### CODE REVIEW

File: Product/Production/Gateway/CONNECTCommonWeb/src/main/java/gov/hhs/fha/nhinc/common/propaccessor/NhincComponentPropAccessor.java [hunk 1]
```diff
@@ -26,6 +26,11 @@
  */
 package gov.hhs.fha.nhinc.common.propaccessor;
 
+import static gov.hhs.fha.nhinc.nhinclib.NhincConstants.ADAPTER_PROPERTY_FILE_NAME;
+import static gov.hhs.fha.nhinc.nhinclib.NhincConstants.AUDIT_LOGGING_PROPERTY_FILE;
+import static gov.hhs.fha.nhinc.nhinclib.NhincConstants.GATEWAY_PROPERTY_FILE;
+
+//import gov.hhs.fha.nhinc.common.exchangemanagement.SimpleExchangeManagementResponseMessageType;
```
Comment (TabassumJafri) [inline Product/Production/Gateway/CONNECTCommonWeb/src/main/java/gov/hhs/fha/nhinc/common/propaccessor/NhincComponentPropAccessor.java:33]: Remove this import

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/PropertyServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+/**
+ * @author ptambellini
+ *
+ */
+
+@Service
+public class PropertyServiceImpl implements PropertyService {
+    private static final Logger LOG = LoggerFactory.getLogger(PropertyServiceImpl.class);
+    private static final WebServiceProxyHelper oProxyHelper = new WebServiceProxyHelper();
+    private static CONNECTClient<NhincComponentPropAccessorPortType> client = null;
```
Comment (plobre) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/PropertyServiceImpl.java:65]: If we are going to be saving the client, we need to reuse it appropriately. Additionally, we may not be able to cache it at all due to the fact that the URL may change from inside the connection manager an will not update if it is cached.

**Review by plobre — CHANGES_REQUESTED** [review]

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/PropertyServiceImpl.java [hunk 3]
```diff
@@ -116,12 +116,10 @@ private boolean fileNameChk(String file) {
     }
 
     private static CONNECTClient<NhincComponentPropAccessorPortType> getClient() throws Exception {
-        if (null == client) {
-            String url = oProxyHelper.getAdapterEndPointFromConnectionManager(PROPERTIES_SERVICE_NAME);
-            ServicePortDescriptor<NhincComponentPropAccessorPortType> portDescriptor = new NhincComponentPropAccessorPortDescriptor();
-            client = CONNECTCXFClientFactory.getInstance().getCONNECTClientUnsecured(portDescriptor, url,
-                new AssertionType());
-        }
+        String url = oProxyHelper.getAdapterEndPointFromConnectionManager(PROPERTIES_SERVICE_NAME);
```
Comment (vdmehta06) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/PropertyServiceImpl.java]: can url be null? or does the method getting called handling that check?

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/PropertyServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                return response.getPropertyList();
+            } catch (Exception e) {
+                LOG.error("error during Propery List: {}", e.getLocalizedMessage(), e);
+            }
+        }
+        return new ArrayList<>();
+    }
+
+    private boolean fileNameChk(String file) {
+        boolean result = false;
```
Comment (vdmehta06) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/PropertyServiceImpl.java:110]: you might not need the result object. You can return the result directly.

**Pull request closed without merging**